src/api/helpers.ts

```diff
@@ -13,6 +17,84 @@ export async function getAuthenticationHeaders(options?: RequestInit): Promise<H
   return {}
 }
 
+interface JWTToken {
+  exp: number
+  fresh: false
+  iat: number
+  identity: string
+  jti: string
+  nbf: number
+  type: string
+}
+
+/**
+ * Implements a retry-pattern for http calls.
+ * Ignores native/v1/refresh_access_token.
+ *
+ * First decodes the local access token:
+ * on success: continue to the call
+ * on error (401): try to refresh the access token
+ * on error (other): propagates error
+ */
+export const retryableFetch = async (url: string, options: RequestInit): Promise<Response> => {
```

Comment:
c'est pas tellement retry-fetch non ? c'est plus refresh token then fetch

Reply:
freshFetch ?

Reply:
Cette fonction a eu plusieurs vies avant sa forme actuelle. to cleanup !

Reply:
renommé en getValidTokenAndFetch